scopes/preview/preview/preview.main.runtime.tsx

```diff
@@ -600,12 +601,37 @@ export class PreviewMain {
       this.executionRefs.set(ctxId, new ExecutionRef(context));
     });
 
-    const previewRuntime = await this.writePreviewRuntime(context);
+    const previewRuntime = await this.writePreviewEntry();
     const linkFiles = await this.updateLinkFiles(context.components, context);
 
     return [...linkFiles, previewRuntime];
   }
 
+  private writePreviewEntry() {
+    const uiPathFromBvm = getAspectDirFromBvm(UIAspect.id);
+    const artifactPath = PreBundlePreviewTask.getArtifactDirectory();
+    const previewPreBundlePath = join(uiPathFromBvm, artifactPath);
+    const manifestPath = join(previewPreBundlePath, 'asset-manifest.json');
+    const manifest = readJsonSync(manifestPath);
+    const imports = manifest.entrypoints
+      .map((entry: string) =>
+        entry.endsWith('.js')
+          ? `import { run } from '@teambit/ui/${artifactPath}/${entry}';`
+          : `import '@teambit/ui/${artifactPath}/${entry}';`
+      )
+      .join('\n');
+
+    const name = this.workspace?.name || 'workspace';
```

Comment:
We need to see what happens on bare scope with this (we just did some special impl for bare scope for the ui pre-bundle. see here - https://github.com/teambit/bit/pull/8028
(but maybe we can find a simpler solution)